RevenueCatUI/Templates/V2/Components/Text/TextComponentViewModel.swift

```diff
@@ -84,6 +88,29 @@ class TextComponentViewModel {
         apply(style)
     }
 
+    private static func fontWeightFrom(integer weight: Int) -> PaywallComponent.FontWeight {
+        let clampedWeight = max(100, min(weight, 900))
+
+        switch clampedWeight {
+        case 100: return .thin
+        case 200: return .extraLight
+        case 300: return .light
+        case 400: return .regular
+        case 500: return .medium
+        case 600: return .semibold
+        case 700: return .bold
+        case 800: return .extraBold
+        case 900: return .black
+
+        default:
+            let availableWeights = [100, 200, 300, 400, 500, 600, 700, 800, 900]
+            let closest = availableWeights.reduce(availableWeights.first!) { currentClosest, candidate in
+                abs(candidate - weight) < abs(currentClosest - weight) ? candidate : currentClosest
+            }
+            return Self.fontWeightFrom(integer: closest)
+        }
+    }
```

Comment:
I understand what we need here, but it's still painful to see that we need to do this 😢